Unable to build

Building on Ubuntu 19.10
Compiler: gcc (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
Insured all submodules have been updated recursively.

I just pulled the latest code, and get a build failure:

[ 59%] Building C object vendor/couchbase-lite-core/vendor/mbedtls/library/CMakeFiles/mbedx509.dir/x509_create.c.o
In file included from /tmp/2/couchbase-lite-C/src/CBLDatabase_Internal.hh:24,
                 from /tmp/2/couchbase-lite-C/src/CBLBlob_Internal.hh:22,
                 from /tmp/2/couchbase-lite-C/src/CBLBlob.cc:19:
/tmp/2/couchbase-lite-C/src/Listener.hh: In member function ‘void cbl_internal::ListenersBase::clear()’:
/tmp/2/couchbase-lite-C/src/Listener.hh:117:30: error: ‘void CBLListenerToken::removed()’ is protected within this context
  117 |                 tok->removed();
      |                              ^
/tmp/2/couchbase-lite-C/src/Listener.hh:57:10: note: declared protected here
   57 |     void removed() {
      |          ^~~~~~~
In file included from /tmp/2/couchbase-lite-C/src/CBLDatabase_Internal.hh:24,
                 from /tmp/2/couchbase-lite-C/src/CBLDatabase.cc:19:
/tmp/2/couchbase-lite-C/src/Listener.hh: In member function ‘void cbl_internal::ListenersBase::clear()’:
/tmp/2/couchbase-lite-C/src/Listener.hh:117:30: error: ‘void CBLListenerToken::removed()’ is protected within this context
  117 |                 tok->removed();
      |                              ^
/tmp/2/couchbase-lite-C/src/Listener.hh:57:10: note: declared protected here
   57 |     void removed() {
      |          ^~~~~~~
[ 59%] Building CXX object CMakeFiles/CouchbaseLiteCStatic.dir/src/ConflictResolver.cc.o
In file included from /tmp/2/couchbase-lite-C/src/CBLDatabase_Internal.hh:24,
                 from /tmp/2/couchbase-lite-C/src/CBLQuery_Internal.hh:9,
                 from /tmp/2/couchbase-lite-C/src/CBLQuery.cc:20:
/tmp/2/couchbase-lite-C/src/Listener.hh: In member function ‘void cbl_internal::ListenersBase::clear()’:
/tmp/2/couchbase-lite-C/src/Listener.hh:117:30: error: ‘void CBLListenerToken::removed()’ is protected within this context
  117 |                 tok->removed();
      |                              ^
/tmp/2/couchbase-lite-C/src/Listener.hh:57:10: note: declared protected here
   57 |     void removed() {
      |          ^~~~~~~
In file included from /tmp/2/couchbase-lite-C/src/CBLDatabase_Internal.hh:24,
                 from /tmp/2/couchbase-lite-C/src/CBLDocument_Internal.hh:22,
                 from /tmp/2/couchbase-lite-C/src/CBLDocument.cc:21:
/tmp/2/couchbase-lite-C/src/Listener.hh: In member function ‘void cbl_internal::ListenersBase::clear()’:
/tmp/2/couchbase-lite-C/src/Listener.hh:117:30: error: ‘void CBLListenerToken::removed()’ is protected within this context
  117 |                 tok->removed();
      |                              ^
/tmp/2/couchbase-lite-C/src/Listener.hh:57:10: note: declared protected here
   57 |     void removed() {
      |          ^~~~~~~
In file included from /tmp/2/couchbase-lite-C/src/CBLBase.cc:21:
/tmp/2/couchbase-lite-C/src/Listener.hh: In member function ‘void cbl_internal::ListenersBase::clear()’:
/tmp/2/couchbase-lite-C/src/Listener.hh:117:30: error: ‘void CBLListenerToken::removed()’ is protected within this context
  117 |                 tok->removed();
      |                              ^
/tmp/2/couchbase-lite-C/src/Listener.hh:57:10: note: declared protected here
   57 |     void removed() {
      |          ^~~~~~~
[ 62%] Building C object vendor/couchbase-lite-core/vendor/mbedtls/library/CMakeFiles/mbedx509.dir/x509_crl.c.o
make[2]: *** [CMakeFiles/CouchbaseLiteCStatic.dir/build.make:76: CMakeFiles/CouchbaseLiteCStatic.dir/src/CBLBlob.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 62%] Building CXX object vendor/couchbase-lite-core/CMakeFiles/LiteCoreStatic.dir/C/c4BlobStore.cc.o
In file included from /tmp/2/couchbase-lite-C/src/CBLDatabase_Internal.hh:24,
                 from /tmp/2/couchbase-lite-C/src/CBLReplicatorConfig.hh:22,
                 from /tmp/2/couchbase-lite-C/src/CBLReplicator_Internal.hh:20,
                 from /tmp/2/couchbase-lite-C/src/CBLReplicator.cc:20:
/tmp/2/couchbase-lite-C/src/Listener.hh: In member function ‘void cbl_internal::ListenersBase::clear()’:
/tmp/2/couchbase-lite-C/src/Listener.hh:117:30: error: ‘void CBLListenerToken::removed()’ is protected within this context
  117 |                 tok->removed();
      |                              ^
/tmp/2/couchbase-lite-C/src/Listener.hh:57:10: note: declared protected here
   57 |     void removed() {
      |          ^~~~~~~
make[2]: *** [CMakeFiles/CouchbaseLiteCStatic.dir/build.make:89: CMakeFiles/CouchbaseLiteCStatic.dir/src/CBLDatabase.cc.o] Error 1

@snej removed() in Listener.hh should probably be public instead of protected

Unfortunately once I got pass that, I am getting some linker errors now:
```
/usr/bin/ld: ../libCouchbaseLiteC.so: undefined reference to `litecore::crypto::TLSLogDomain'
/usr/bin/ld: ../libCouchbaseLiteC.so: undefined reference to `litecore::crypto::Cert::Cert(fleece::slice)'

```

--- a/CBL_C.xcodeproj/xcshareddata/xcschemes/CBL_C Framework.xcscheme
+++ b/CBL_C.xcodeproj/xcshareddata/xcschemes/CBL_C Framework.xcscheme
@@ -27,15 +27,6 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
-      <MacroExpansion>
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "27984E092249A126000FE777"
-            BuildableName = "CouchbaseLite.framework"
-            BlueprintName = "CBL_C Framework"
-            ReferencedContainer = "container:CBL_C.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
       <Testables>
       </Testables>
    </TestAction>

--- a/CBL_C.xcodeproj/xcshareddata/xcschemes/CBL_Tests.xcscheme
+++ b/CBL_C.xcodeproj/xcshareddata/xcschemes/CBL_Tests.xcscheme
@@ -27,20 +27,7 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
-      <MacroExpansion>
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "27B61DA721D6E49D0027CCDB"
-            BuildableName = "CBL_Tests"
-            BlueprintName = "CBL_Tests"
-            ReferencedContainer = "container:CBL_C.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
       <TestPlans>
-         <TestPlanReference
-            reference = "container:Xcode/CBL_Tests.xctestplan"
-            default = "YES">
-         </TestPlanReference>
       </TestPlans>
    </TestAction>
    <LaunchAction

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,8 +128,8 @@ install (TARGETS CouchbaseLiteC
 set(CBL_LIBRARIES_PRIVATE  ${WHOLE_LIBRARY_FLAG}
                             CouchbaseLiteCStatic
                             FleeceStatic
-                           ${NO_WHOLE_LIBRARY_FLAG}
                             LiteCoreStatic
+                           ${NO_WHOLE_LIBRARY_FLAG}
                             LiteCoreWebSocket)
 
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "^armv[67]")

--- a/include/cbl/CBL_Compat.h
+++ b/include/cbl/CBL_Compat.h
@@ -36,7 +36,7 @@
     #define _cbl_nonnull            _In_
     #define _cbl_returns_nonnull    _Ret_notnull_
     #define _cbl_warn_unused        _Check_return_
-    #define _cbl_deprecated(MSG,REPLACEMENT)
+    #define _cbl_deprecated
 #else
     #define CBLINLINE            inline
     #define _cbl_returns_nonnull __attribute__((returns_nonnull))

--- a/jenkins/jenkins_unix.sh
+++ b/jenkins/jenkins_unix.sh
@@ -4,8 +4,7 @@ set -e
 shopt -s extglob dotglob
 
 function build_xcode {
-    xcodebuild -project CBL_C.xcodeproj -configuration Debug-EE -derivedDataPath ios -scheme "CBL_C framework" -sdk iphonesimulator CODE_SIGNING_ALLOWED=NO
-    popd
+    xcodebuild -project CBL_C.xcodeproj -configuration Debug-EE -derivedDataPath ios -scheme "CBL_C Framework" -sdk iphonesimulator CODE_SIGNING_ALLOWED=NO
 }
 
 

--- a/src/CBLReplicatorConfig.hh
+++ b/src/CBLReplicatorConfig.hh
@@ -76,7 +76,7 @@ namespace cbl_internal {
         virtual C4Database* otherLocalDB() const override       {return _db->_getC4Database();}
 
     private:
-        Retained<CBLDatabase> _db;
+        fleece::Retained<CBLDatabase> _db;
     };
 #endif
 }

